fault/tester/sequence_tester.py

```diff
@@ -0,0 +1,34 @@
+from fault.tester.staged_tester import Tester
+
+
+class SequenceTester(Tester):
+    def __init__(self, circuit, driver, monitor, sequence, clock=None):
+        super().__init__(circuit, clock)
+        self.driver = driver
+        self.driver.set_tester(self)
+        self.monitor = monitor
+        self.monitor.set_tester(self)
+        self.sequence = sequence
+
+    def _compile_sequences(self):
+        for item in self.sequence:
+            self.driver.consume(*item)
+            if self.clock is None:
+                self.eval()
+            else:
+                self.step(2)
+            self.monitor.observe(*item)
+
+    def _compile(self, target="verilator", **kwargs):
+        self._compile_sequences()
+        super()._compile(target, **kwargs)
+
+
+class Driver:
```

Comment:
maybe should add `abstractmethod`s `consume` and `observe` to `Driver` and `Monitor` classes respectively.

On that note, open to new names on these functions. Idk if UVM defines these.

Reply:
observe seems fine for monitor, I would maybe swap `drive` for `consume`? 

I think one pattern is: a driver "drives" values and a monitor "monitors" values, the problem there is conflates the object/entity with the action since the words are similar (that's why observe and monitor seems nice), not sure if there's a synonym for "drive" we can use that captures the same relationship as observe does

Reply:
Agreed. Ironically, perhaps `produce` is better than `consume` 😂 ? Another option is `lower`?

Reply:
I like `lower` for now, it's fairly explicit in the role of the method.